modules/auxiliary/gather/ldap_query.rb

```diff
@@ -0,0 +1,313 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::LDAP
+  require 'json'
+  require 'yaml'
+
+  def initialize(info = {})
+    filename = 'ldap_queries_default.yaml'
+    user_config_file = File.join(::Msf::Config.get_config_root.to_s, filename)
+    unless File.exist?(user_config_file)
+      # If the user config file doesn't exist, then initialize it with the contents of the default one.
+      default_config_file = File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'ldap_query', filename)
+      FileUtils.cp(default_config_file, user_config_file)
+    end
```

Comment:
This feels clunky; What happens if we update the example ldap query files in the metasploit codebase? The user would never have the newest version available to them unless they manually delete the old version that was implicitly copied across to their config folder

Reply:
So this was taken from a chat @smcintyre-r7 and I were having. The idea was that the user wouldn't update the LDAP query file in the Metasploit folder but rather in the config folder itself and that way they could adjust it to include their own default set of queries if needed.

That being said you are correct, if we were to release a new version of the default query file then with the current setup of the code, then technically this wouldn't be updated.

I don't know if we have a good middle ground on this to be honest without the risk of overwriting someone's changes to the file.

Reply:
How about merging the queries loaded from both? Then if we shipped a new file with a new query it'd be available but the users modifications to any of the existing ones along with custom queries (as long as the names are unique) would be kept.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/16598/commits/e8bf8a173a3d548090816b1854f704cc448c2939.